src/contracts/libraries/GPv2TradeExecution.sol

```diff
@@ -34,6 +38,10 @@ library GPv2TradeExecution {
     /// @dev Executes the trade's buy amount, transferring it to the trade's
     /// owner from the caller's address.
     function transferBuyAmountToOwner(Data memory trade) internal {
-        trade.buyToken.safeTransfer(trade.owner, trade.buyAmount);
+        if (address(trade.buyToken) == BUY_ETH_ADDRESS) {
+            payable(trade.owner).transfer(trade.buyAmount);
```

Comment:
Does the EVM allow you to just transfer people's eth using offline signature?

Reply:
No, this is transferring from the settlement contract to an EOA.